### PR TITLE
Feature/add network part

### PR DESCRIPTION
**Pull request by p0l0satik**

### CODE REVIEW

File: loader/dataset.py [hunk 1]
```diff
@@ -25,15 +25,6 @@ def __len__(self):
...
-                cloud["points"],
-                cloud["labels"],
-                cloud["proj_idx"],
-            )
-
-        return SphericalProjectionReturnValue(
-            cloud["proj_points"],
-            cloud["proj_labels"],
-        )
+            return cloud["proj_points"], cloud["proj_labels"],cloud["points"],cloud["labels"], cloud["proj_idx"]
```
Comment (kirill-ivanov-a) [inline loader/dataset.py:28]: There is not a space after the comma everywhere. All these minor inaccuracies in syntax can be corrected by a linter :)

File: loader/dataset.py [hunk 2]
```diff
@@ -25,15 +25,6 @@ def __len__(self):
...
+        return  cloud["proj_points"],cloud["proj_labels"],
```
Comment (kirill-ivanov-a) [inline loader/dataset.py:29]: why extra comma?

File: network/common_blocks.py [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+
+    optimizer = torch.optim.AdamW(model.parameters())
+    scheduler = torch.optim.lr_scheduler.ReduceLROnPlateau(
+        optimizer,
+        mode="min",
+        factor=config.factor,
+        patience=config.patience,
+        threshold=config.threshold,
+    )
+    return model, optimizer, scheduler
```
Comment (kirill-ivanov-a) [inline network/common_blocks.py:39]: The return type doesn't seem to match the function name

File: network/common_blocks.py [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+
+
+def one_epoch(config, loader, optimizer, model, criterion=None, train_mode=True):
```
Comment (kirill-ivanov-a) [inline network/common_blocks.py:42]: The function name usually reflects some action

File: network/common_blocks.py [hunk 5]
```diff
@@ -0,0 +1,182 @@
...
+            )
+
+            validation_logger.log_one_step(
+                curr_time, batch_iou, precision, recall, dice, pixel_accuracy
+            )
+    validation_logger.log_final()
+
+
+def train(
+    tl,
```
Comment (kirill-ivanov-a) [inline network/common_blocks.py:114]: What do tl and vl mean? It's better to call it more expressively

File: network/config.py [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+        self.random_seed = config_file["other"]["random_seed"]
+        self.curr_time = datetime.now().strftime("%Y%m%d_%H%M%S")
+
+    def prepare(self):
+        self.checkpoint_dir = self.checkpoint_dir / f"{self.run_name}_{self.curr_time}"
+        # create a directory for checkpoints if not exists
+        self.checkpoint_dir.mkdir(parents=True, exist_ok=True)
+
+        if self.use_wandb:
+            # TODO: log all parameters
```
Comment (kirill-ivanov-a) [inline network/config.py:64]: Before merging, TODO must either be removed or implemented

File: loader/dataset.py [hunk 7]
```diff
@@ -25,15 +25,6 @@ def __len__(self):
     def __getitem__(self, idx):
         cloud = np.load(self.load_path / "scan_{0:06d}.npz".format(idx))
         if self.return_orig_points:
-            return SphericalProjectionReturnValue(
```
Comment (kirill-ivanov-a) [inline loader/dataset.py:28]: Why did you decide to abandon the class for the return value?

File: network/common_blocks.py [hunk 8]
```diff
@@ -0,0 +1,182 @@
+import torch
```
Comment (kirill-ivanov-a) [inline network/common_blocks.py:1]: It seems to me that you can create a class that will perform training, validation, and so on. In order not to pass the same parameters everywhere

**Review by kirill-ivanov-a — CHANGES_REQUESTED** [review]

File: loader/dataset.py (hunk 7 above)
Comment (p0l0satik) [reply loader/dataset.py:28]: It was done so torch Dataloader does not support torch Dataset to return Classes by default. https://stackoverflow.com/questions/72867109/what-is-pytorch-dataset-supposed-to-return https://github.com/pytorch/pytorch/blob/main/torch/utils/data/_utils/collate.py#L85-L130 So I decided to revert and leave as it was

**Review by kirill-ivanov-a — APPROVED** [review]

**Pull request merged**